### PR TITLE
stac-1988 fix missing saved search

### PR DESCRIPTION
**Pull request by vishalxebia**

### CODE REVIEW

File: utils/splunk/splunk_helper.py [hunk 1]
```diff
@@ -117,6 +117,8 @@ def _do_post(self, path, payload, request_timeout_seconds):
         try:
             resp.raise_for_status()
         except HTTPError as error:
-            self.log.error("Received error response with status {} and body {}".format(resp.status_code, resp.content))
-            raise error
+            self.log.warn("Received response with status {} and body {}".format(resp.status_code, resp.content))
```
Comment (TizianoPerrucci) [inline utils/splunk/splunk_helper.py:123]: should be a `log.error` if `splunk_ignore_config` false

**Review by TizianoPerrucci — CHANGES_REQUESTED** [review]

File: utils/splunk/splunk_helper.py (hunk 1 above)
Comment (vishalxebia) [reply utils/splunk/splunk_helper.py:123]: ok, I will remove the warning and simply put the error in if `splunk_ignore_config` false block.

File: utils/splunk/splunk_helper.py [hunk 2]
```diff
@@ -99,16 +99,16 @@ def dispatch(self, saved_search, splunk_user, splunk_app, parameters):
...
+        return response_body.get("sid")
 
     def _do_get(self, path, request_timeout_seconds, verify_ssl_certificate):
         url = "%s%s" % (self.instance_config.base_url, path)
         response = self.requests_session.get(url, timeout=request_timeout_seconds, verify=verify_ssl_certificate)
         response.raise_for_status()
         return response
 
-    def _do_post(self, path, payload, request_timeout_seconds):
+    def _do_post(self, path, payload, request_timeout_seconds, splunk_ignore_config='true'):
```
Comment (whazenberg) [inline utils/splunk/splunk_helper.py]: why is the `splunk_ignore_config` stringly typed? Can it be a `True`?

Comment (whazenberg) [inline utils/splunk/splunk_helper.py]: I interpret the name `splunk_ignore_config` that there is a specific config that ignores something(s). However, it is only a single true/false variable. Can we rename the variable to something more similar to what the user setting is, `ignore_saved_search_errors`?

File: utils/splunk/splunk_helper.py [hunk 3]
```diff
@@ -117,6 +117,14 @@ def _do_post(self, path, payload, request_timeout_seconds):
         try:
             resp.raise_for_status()
         except HTTPError as error:
+            if not splunk_ignore_config:
+                self.log.error("Received error response with status {} and body {}".format(resp.status_code, resp.content))
+                raise error
```
Comment (whazenberg) [inline utils/splunk/splunk_helper.py:122]: can we create a test in integrations-core for this? i.e. an exception needs to be thrown when `splunk_ignore_config=False`

**Review by whazenberg — CHANGES_REQUESTED** [review]

File: utils/splunk/splunk_helper.py [hunk 4]
```diff
@@ -122,9 +122,11 @@ def _do_post(self, path, payload, request_timeout_seconds, splunk_ignore_config=
                 raise error
             self.log.warn("Received response with status {} and body {}".format(resp.status_code, resp.content))
         except Timeout as error:
-            self.log.error("Got a timeout error")
-            raise error
+            if not splunk_ignore_config:
+                self.log.error("Got a timeout error")
```
Comment (whazenberg) [inline utils/splunk/splunk_helper.py:126]: do we want to log anything when we ignore the errors? It may be useful to have any clue in the logs

**Review by whazenberg — CHANGES_REQUESTED** [review]

**Review by TizianoPerrucci — APPROVED** [review]

File: utils/splunk/splunk_helper.py (hunk 2 above)
Comment (vishalxebia) [reply utils/splunk/splunk_helper.py]: Done.

File: utils/splunk/splunk_helper.py (hunk 4 above)
Comment (vishalxebia) [reply utils/splunk/splunk_helper.py:126]: logged a warning now.

File: utils/splunk/splunk_helper.py (hunk 2 above)
Comment (vishalxebia) [reply utils/splunk/splunk_helper.py]: yeah it's done now.

**Review by whazenberg — APPROVED** [review]

**Pull request merged**